src/openloco/localisation/languagefiles.cpp

```diff
@@ -0,0 +1,274 @@
+                    {
+                        *out = colour->second;
+                        out++;
+                    }
+                    else
+                    {
+                        printf("%.*s\n", (int)commands[1].length(), commands[1].data());
+                    }
+                }
+                else if (commands[0] == "MOVE_X")
+                {
+                    *out++ = (char)control_codes::move_x;
+                    uint8_t pixels_to_move_by = std::atoi(commands[1].data());
+                    *out++ = pixels_to_move_by;
+                }
+                else if (commands[0] == "NEWLINE")
+                {
+                    if (commands.size() == 1)
+                    {
+                        *out++ = (char)control_codes::newline;
+                    }
+                    else if (commands.size() == 3)
+                    {
+                        *out++ = (char)control_codes::newline_x_y;
+
+                        uint8_t x_pixels_to_move_by = std::atoi(commands[1].data());
+                        *out++ = x_pixels_to_move_by;
+
+                        uint8_t y_pixels_to_move_by = std::atoi(commands[2].data());
+                        *out++ = y_pixels_to_move_by;
+                    }
+                }
+                else
+                {
+                    printf("%.*s\n", (int)commands[0].length(), commands[0].data());
+                }
+
+                continue;
+            }
+            else
+            {
+                *out = readChar;
+                out++;
+            }
+
+            if (readChar == '\0')
+                break;
+        }
+
+        return str;
+    }
+
+    static bool stringIsBuffer(int id)
+    {
+        return id == string_ids::buffer_337 || id == string_ids::buffer_338 || id == string_ids::buffer_1250 || id == string_ids::preferred_currency_buffer || id == string_ids::buffer_1719
+            || id == string_ids::buffer_2039 || id == string_ids::buffer_2040 || id == string_ids::buffer_2042 || id == string_ids::buffer_2045;
+    }
+
+    static bool loadLanguageStringTable(std::string languageFile)
+    {
+        try
+        {
+            YAML::Node node = YAML::LoadFile(languageFile);
+            node = node["strings"];
+
+            for (YAML::const_iterator it = node.begin(); it != node.end(); ++it)
+            {
+                int id = it->first.as<int>();
+                if (stringIsBuffer(id))
+                    continue;
+
+                std::string new_string = it->second.as<std::string>();
+                char* processed_string = readString(new_string.data(), new_string.length());
+
+                if (processed_string != nullptr)
+                    _strings[id] = processed_string;
+            }
+
+            return true;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << e.what() << "\n";
+            return false;
+        }
+    }
+
+    bool loadLanguageFile()
```

Comment:
`loadLanguageFile` doesn't sound like it returns a boolean. An exception would be more appropriate.